### ISSUE TITLE
Channel export fails with System.IndexOutOfRangeException

### ISSUE THREAD

**Issue by TomArrow** [Tyrrrz/DiscordChatExporter#935]
### Version

v2.36.0

### Flavor

GUI (Graphical User Interface)

### Export format

HTML

### Details

I tried to export a channel in HTML (Dark) without Limits and with downloading media activated.

At some point, an error appeared. I cannot actually mark and copy the error so here's the screenshot of the whole error message, but it starts with System.IndexOutOfRangeException: Index was outside the bounds of the array at DiscordChatExporter.Core.Exporting.Writers.Html.MessageGroupTemplate.ExecuteAsync() in ....

![2022-09-18 01_30_18-DiscordChatExporter v2 36 0](https://user-images.githubusercontent.com/6640421/190879656-99c27de5-4286-408f-99e1-f05df63b0746.png)

![2022-09-18 01_30_27-DiscordChatExporter v2 36 0](https://user-images.githubusercontent.com/6640421/190879657-bc608b36-2413-4d59-9d66-ef5bcd836dbc.png)

![2022-09-18 01_30_35-DiscordChatExporter v2 36 0](https://user-images.githubusercontent.com/6640421/190879659-fa958a15-4c0e-4bf3-9a5f-0d20a70b5375.png)

![2022-09-18 01_30_41-DiscordChatExporter v2 36 0](https://user-images.githubusercontent.com/6640421/190879661-f2563be3-f056-40d3-84d9-695b19a616f7.png)

### Steps to reproduce

I don't know. Some channels work fine but on 2 I had this issue so far. Maybe it's some particular kind of content that is parsed wrongly or who knows what.

**Comment by TomArrow** [comment]
Oh and if it helps, it seems it happens at the same place every time if i retry. The output .html file ends up the same size.

**Comment by Tyrrrz** [comment]
Can you point exact message that triggers this error?

**Comment by TomArrow** [comment]
Not sure how I can tell, but this is the area where it stops roughly:
![bug](https://user-images.githubusercontent.com/6640421/190881653-4f4532e5-99b1-4446-bc1b-a2a4d37399ad.png)

The .html file cuts off exactly at the "for some reason xD" comment, like this: 
```<span class=chatlog__markdown-preserve>for some reas```

That's the end of the .html file that's generated. But I figure it might just be that some write buffer wasn't fully flushed and it was actually later, who knows.

In another file it happens here:
![bug2](https://user-images.githubusercontent.com/6640421/190881811-140e74bd-21b7-4f44-8f41-ea0af6744697.png)

Right after "Game is dead anyway" it cuts off mid-HTML like this:
```alt=Avatar loading=lazy></div><div cl```

Maybe something related to usage of bot commands?

**Comment by TomArrow** [comment]
I just tried exporting one of those channels as JSON and that seems to have finished without problems.

**Comment by LuminairPrime** [comment]
Can confirm, I just got this too! There doesn't seem to be anything specifically interesting about the message before or after it cut off.

**Comment by Tyrrrz** [comment]
I have a hunch that it could be messages without text content. Can you try exporting just that one message (using date ranges) and see if it fails?

**Comment by LuminairPrime** [comment]
> Can you try exporting just that one message (using date ranges) and see if it fails?

I think I've got it pinned down! I can get an exception filtering for one user, in one channel, in a 2 day range, in 2019. The only unique thing visible in discord is the user giving the server a "boost". I then tried filtering for a different user in 2019 who also boosted the server, and that also caused the exception. But when I filtered for a different boost in 2022, that DID export, albeit incompletely with the number of boosts after the username and before the date, and no message body. 

In conclusion, the different behaviors on different dates seems to imply that Discord is storing older boost messages in a different format than newer boost messages, and the app isn't tolerating the older format. I hope that helps a lot!!! Keep up the good work :)

PS: Is it a known issue that having a date range of the same day finds nothing? So: exporting 6/4/2019 to 6/4/2019 works (no messages found), and exporting 6/5/2019 to 6/5/2019 works (no messages found), but exporting 6/4/2019 to 6/5/2019 throws an exception.

**Comment by Tyrrrz** [comment]
Please test out the CI build once it completes

**Comment by Tyrrrz** [comment]
> Is it a known issue that having a date range of the same day finds nothing? So: exporting 6/4/2019 to 6/4/2019 works (no messages found), and exporting 6/5/2019 to 6/5/2019 works (no messages found)

It's not an issue, it's logical since 6/4/2019 refers to 6/4/2019 00:00

**Comment by TomArrow** [comment]
Can confirm it works now with the CI build, thanks!
